### ISSUE TITLE
[bug] Per-package settings override seems impossible for settings_target

### ISSUE THREAD

**Issue by fschoenm** [conan-io/conan#14521]
### Environment details

* Operating System+version: Ubuntu 2023.04
* Compiler+version: n/a
* Conan version: 2.0.10
* Python version: 3.11

### Steps to reproduce

In #14366, I was told to manually set my desired cross compiler (e.g. in a profile) by using a per-package setting. But it doesn't seem to work. (Or I don't know how to do it correctly.)

Background: I have two cross-compiler packages with different `settings_target` (see at the bottom for details):
* `6657...` for x64
* `fc10...` for armv8.2 (should be armv8 but that's not relevant for this issue)

I want to build for `armv8.3` now using the cross-compiler package for `armv8.2`:
```
$ conan install --tool-requires 'gcc/10.3.0' -s:h os.glibc=2.31 -s:h arch=armv8.3 -s:h 'gcc*:arch=armv8.2
======== Input profiles ========
Profile host:
[settings]
arch=armv8.3
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=10
os=Linux
os.glibc=2.31
gcc*:arch=armv8.2
[conf]
tools.cmake.cmaketoolchain:generator=Ninja

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=10
os=Linux
os.glibc=2.35
[conf]
tools.cmake.cmaketoolchain:generator=Ninja

======== Computing dependency graph ========
Graph root
    cli
Build requirements
    crosstools_ng/1.25.0#de2f75827d46efa515a1b7c8b56d2eb8 - Cache
    gcc/10.3.0#862048caa4fdffcf6da5c0ef1135519d - Cache
Resolved version ranges
    crosstools_ng/[~1]: crosstools_ng/1.25.0
======== Computing necessary packages ========
crosstools_ng/1.25.0: Checking glibc compatibility for 2.35 against [None, 2.27, 2.31, 2.35]
crosstools_ng/1.25.0: Compatible package ID 533423301a96984b0e2a24df835df2e79d0af584 equal to the default package ID
crosstools_ng/1.25.0: Checking 2 compatible configurations:
crosstools_ng/1.25.0: '6899a106319b7ceb6f60c939cee12a09e05e8b9c': os.glibc=2.27
crosstools_ng/1.25.0: 'd5ee09c60d4f637985b270e11a6fcf9b30141665': os.glibc=2.31
crosstools_ng/1.25.0: Main binary package '533423301a96984b0e2a24df835df2e79d0af584' missing. Using compatible package 'd5ee09c60d4f637985b270e11a6fcf9b30141665'
gcc/10.3.0: Checking glibc compatibility for 2.35 against [None, 2.27, 2.31, 2.35]
gcc/10.3.0: Compatible package ID af219bfa8fea1fe3f8b30b78530084c06cca8f37 equal to the default package ID
gcc/10.3.0: Checking 2 compatible configurations:
gcc/10.3.0: '5f450740d5a7cd89ecf89c4bacb297df3cec3ef8': os.glibc=2.27
gcc/10.3.0: '48fea60adbddacd667ab82ad7618955f57bd4f3a':
Build requirements
    gcc/10.3.0#862048caa4fdffcf6da5c0ef1135519d:af219bfa8fea1fe3f8b30b78530084c06cca8f37 - Missing
Skipped binaries
    crosstools_ng/1.25.0

======== Installing packages ========
ERROR: Missing binary: gcc/10.3.0:af219bfa8fea1fe3f8b30b78530084c06cca8f37

gcc/10.3.0: WARN: Can't find a 'gcc/10.3.0' package binary 'af219bfa8fea1fe3f8b30b78530084c06cca8f37' for the configuration:
[settings]
arch=x86_64
os=Linux
os.glibc=2.35
[settings_target]
arch=armv8.3
os=Linux
os.glibc=2.31

ERROR: Missing prebuilt package for 'gcc/10.3.0'
Check the available packages using 'conan list gcc/10.3.0:* -r=remote'
or try to build locally from sources using the '--build=gcc/10.3.0' argument

More Info at 'https://docs.conan.io/2/knowledge/faq.html#error-missing-prebuilt-package'
```

As you can see all the way at the end (in the profile/settings for gcc), the CLI settings has no impact on the `settings_target` for this tool.

I'm not sure if that's a bug, a missing feature or I'm just using it wrong.

### Logs

```
$ conan list 'gcc:*'
Local Cache
  gcc
    gcc/10.3.0
      revisions
        862048caa4fdffcf6da5c0ef1135519d (2023-08-01 11:39:36 UTC)
          packages
            6657d9078efa15debbbb02fce5f894eb0b55993c
              info
                settings
                  arch: x86_64
                  os: Linux
                  os.glibc: 2.35
                settings_target
                  arch: x86_64
                  os: Linux
                  os.glibc: 2.31
            fc10e41224a798a043cbb3a2a46dee13759695ad
              info
                settings
                  arch: x86_64
                  os: Linux
                  os.glibc: 2.35
                settings_target
                  arch: armv8.2
                  os: Linux
                  os.glibc: 2.31
```

**Comment by fschoenm** [comment]
What might also be interesting (and part of the problem) is, that the configuration for gcc in the above output contains a `[settings_target]` section but I cannot add that section to my profile or the config parser complains.

**Comment by memsharded** [comment]
Thanks for the report @fschoenm 

I am trying to add the computation of per-package settings to apply to ``settings_target``, included in PR https://github.com/conan-io/conan/pull/14441 that contains other ``settings_target`` fix.

> What might also be interesting (and part of the problem) is, that the configuration for gcc in the above output contains a [settings_target] section but I cannot add that section to my profile or the config parser complains.

Yes, this is expected. ``settings_target`` doesn't have a different value per-se, but it is a function of the host-build settings, based on the current context.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/14441 has been merged, it will be in next 2.0.10
